ghost/members-api/lib/repositories/member.js

```diff
@@ -337,12 +337,24 @@ module.exports = class MemberRepository {
                 }
             }
         }
-
-        DomainEvents.dispatch(MemberCreatedEvent.create({
-            memberId: member.id,
-            attribution: data.attribution,
-            source
-        }, eventData.created_at));
+        if (options?.transacting) {
+            // Only dispatch the event after the transaction has finished
+            options.transacting.executionPromise.then(async () => {
+                DomainEvents.dispatch(MemberCreatedEvent.create({
+                    memberId: member.id,
+                    attribution: data.attribution,
+                    source
+                }, eventData.created_at));
+            }).catch(() => {
+                // ignore errors for transaction rollback and don't dispatch event
+            });
```

Comment:
Just so I understand this correctly, what happens when we don't add an empty catch? + we don't have this catch in `linkSubscription`

Reply:
@SimonBackx If we don't catch it then the error floats up in case of a transaction rollback and we get `Unhandled Rejection`. The catch here ensures we don't do anything wrt event if for any reason transaction is not committed, which should ideally only happen for the importer. 
I have extracted this out in a common method now and applied to all usages in member repo, so this should help us cover any suspect event dispatch.